lib/-private/file-path.js

```diff
@@ -0,0 +1,73 @@
+import { parse } from 'node:path';
+
+const DTS = '.d.ts';
+export const GJS = ['.gjs', '.gts'];
+const HBS = '.hbs';
+const HTML = '.html';
+export const SOON_DEPRECATED = ['.js', '.ts'];
+
+function isDTS(ext) {
+  return ext.endsWith(DTS);
+}
+
+function isGlimmerScript(ext) {
+  return GJS.includes(ext);
+}
+
+function isHBS(ext) {
+  return ext === HBS;
+}
+
+function isHTML(ext) {
+  return ext === HTML;
+}
+
+function isScript(ext) {
+  return SOON_DEPRECATED.includes(ext);
+}
```

Comment:
To avoid unintended errors (especially in the absence of TypeScript), I'm performing exact matches instead of partial (what had been done before via `.endsWith()`). It's okay to use `.endsWith()` to match file extensions `*.d.ts` and `*.css.d.ts`.